it/src/test/java/com/google/cloud/teleport/it/spanner/DefaultSpannerResourceManagerTest.java

```diff
@@ -0,0 +1,381 @@
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareCreateInstanceMock();
+    when(spanner.getDatabaseAdminClient().createDatabase(any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    prepareUpdateDatabaseMock();
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTableShouldThrowExceptionWhenSpannerUpdateDatabaseFails()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareCreateInstanceMock();
+    prepareCreateDatabaseMock();
+    when(spanner.getDatabaseAdminClient().updateDatabaseDdl(any(), any(), any(), any()).get())
+        .thenThrow(InterruptedException.class);
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act & assert
+    assertThrows(SpannerResourceManagerException.class, () -> testManager.createTable(statement));
+  }
+
+  @Test
+  public void testCreateTableShouldWorkWhenSpannerDoesntThrowAnyError()
+      throws ExecutionException, InterruptedException {
+    //   arrange
+    prepareCreateInstanceMock();
+    prepareCreateDatabaseMock();
+    prepareUpdateDatabaseMock();
+
+    String statement =
+        "CREATE TABLE Singers (\n"
+            + "  SingerId   INT64 NOT NULL,\n"
+            + "  FirstName  STRING(1024),\n"
+            + "  LastName   STRING(1024),\n"
+            + ") PRIMARY KEY (SingerId)";
+
+    // act
+    testManager.createTable(statement);
+
+    // assert
+    verify(spanner.getDatabaseAdminClient(), times(2))
+        .updateDatabaseDdl(
+            instanceIdCaptor.capture(),
+            databaseIdCaptor.capture(),
+            statementCaptor.capture(),
+            any());
+
+    String actualInstanceId = instanceIdCaptor.getValue();
+    String actualDatabaseId = databaseIdCaptor.getValue();
+    Iterable<String> actualStatement = statementCaptor.getValue();
+
+    assertThat(actualInstanceId).startsWith(TEST_ID);
+    assertThat(actualInstanceId).isNotEqualTo(TEST_ID);
+    assertThat(actualDatabaseId).isEqualTo(TEST_ID);
+    assertThat(actualStatement).containsExactlyElementsIn(ImmutableList.of(statement));
+  }
+
+  @Test
+  public void testWriteSingleRecordShouldWorkWhenSpannerWriteSucceeds()
+      throws ExecutionException, InterruptedException {
+    // arrange
+    prepareCreateInstanceMock();
+    prepareCreateDatabaseMock();
+    prepareUpdateDatabaseMock();
+    testManager.createTable("");
```

Comment:
Since prepareCreateInstanceMock() ... testManager.createTable("")  is repeated from here on, can I move it to setup and have methods that need a different behavior(e.g. throw error for createInstance)  redefine it?